analysisbucket/tests/test_client.py

```diff
@@ -0,0 +1,60 @@
+from analysisbucket.client import AnalysisClient, fill_result
+from numpy.testing import assert_array_equal
+from pandas.util.testing import assert_frame_equal
+
+import numpy as np
+import pandas as pd
+
+
+def test_header_creation():
+    ac = AnalysisClient()
+    md = {k: ord(k) for k in 'abcdef'}
+    h = ac.create_header('testing', uid='fixed', time=0, **md)
+
+    for k, v in md.items():
+        assert h[k] == v
+
+    assert h.name == 'testing'
+    assert h.uid == 'fixed'
+    assert h.time == 0
+
+
+def test_add_result():
+    ac = AnalysisClient()
+
+    md = {k: ord(k) for k in 'abcdef'}
+
+    h = ac.create_header('testing', uid='fixed', time=0, **md)
+    h = h.Document()
+
+    tab = pd.DataFrame({'a': range(15), 'b': 'a'*15})
+    arr = np.arange(15)
+
+    res = ac.add_result(h, tab=tab, arr=arr)
+
+    for d in [res['data'], res.descriptor.data_keys]:
+        for k in ['tab', 'arr']:
+            assert k in d
+    res = fill_result(res)
+
+    assert_array_equal(res.data['arr'], arr)
```

Comment:
You could use all(), I guess, but the advantage of numpy tests is it checks
dtype and so on.
On Sat, Dec 26, 2015 at 1:44 PM Thomas A Caswell <notifications@github.com>
wrote:

> In analysisbucket/tests/test_client.py
> <https://github.com/NSLS-II/analysisstore/pull/3#discussion_r48448981>:
>
> > +    md = {k: ord(k) for k in 'abcdef'}
> > +
> > +    h = ac.create_header('testing', uid='fixed', time=0, **md)
> > +    h = h.Document()
> > +
> > +    tab = pd.DataFrame({'a': range(15), 'b': 'a'*15})
> > +    arr = np.arange(15)
> > +
> > +    res = ac.add_result(h, tab=tab, arr=arr)
> > +
> > +    for d in [res['data'], res.descriptor.data_keys]:
> > +        for k in ['tab', 'arr']:
> > +            assert k in d
> > +    res = fill_result(res)
> > +
> > +    assert_array_equal(res.data['arr'], arr)
>
> No, you get an exception that the truth value of an array is ambiguous.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NSLS-II/analysisstore/pull/3/files#r48448981>.
>